sample/src/main/kotlin/com/slack/circuit/sample/home/HomePresenter.kt

```diff
@@ -80,21 +101,29 @@ class HomePresenter
 @AssistedInject
 constructor(
   @Assisted private val navigator: Navigator,
-  petListPresenterFactory: PetListPresenter.Factory
+  petListFilterPresenterFactory: PetListFilterPresenter.Factory
 ) : Presenter<HomeScreen.State, HomeScreen.Event> {
-  private val petListPresenter = petListPresenterFactory.create(navigator)
+  private val petListFilterPresenter = petListFilterPresenterFactory.create()
 
-  @SuppressLint("FlowOperatorInvokedInComposition")
   @Composable
   override fun present(events: Flow<HomeScreen.Event>): HomeScreen.State {
-    val homeNavState =
-      homeNavPresenter(events.filterIsInstance<HomeScreen.Event.HomeEvent>().map { it.event })
-    val petListState =
-      petListPresenter.present(
-        events.filterIsInstance<HomeScreen.Event.PetListEvent>().map { it.event }
-      )
+    val rememberHomeNavState = remember {
+      events.filterIsInstance<HomeScreen.Event.HomeEvent>().map { it.event }
+    }
+    val homeNavState = homeNavPresenter(rememberHomeNavState)
+
+    val rememberPetListFilterState = remember {
+      events.filterIsInstance<HomeScreen.Event.PetListFilterEvent>().map { it.event }
+    }
+    val petListFilterState = petListFilterPresenter.present(rememberPetListFilterState)
```

Comment:
I wonder if there's a way we can create a helper function for this bit of ceremony 🤔

Reply:
Can we address this in a follow-up?

Reply:
Yah let's file an issue to track so we don't forget